### ISSUE TITLE
Yep it works, but the cheat still doesnt for me (well kinda)

### ISSUE THREAD

**Issue by cooldude315** [AlexBurneikis/SoT-DLC#1]

**Comment by cooldude315** [comment]
So basically i loaded the cheat and the first thing that shows is the crosshair and the fps and dougdruids thing, but im not sure how the ship and player esp is supposed to work

**Comment by AlexBurneikis** [comment]
If you look at your ship, there should be a white dot for it. This only works if you run it while in a game.

**Comment by cooldude315** [comment]
when i try to run it while in game i get this 
![image](https://user-images.githubusercontent.com/117041200/200411741-9628020b-4351-417e-93b8-8aca00ea84b5.png)

**Comment by cooldude315** [comment]
and after trying to re load it for awhile and it laods i get this 
![image](https://user-images.githubusercontent.com/117041200/200412022-36a3f6d3-2c03-4699-8831-a0096dca8d9b.png)

**Comment by AlexBurneikis** [comment]
Windows or steam version?

**Comment by cooldude315** [comment]
> Windows or steam version?

windows

**Comment by AlexBurneikis** [comment]
That could be what's creating the issue, I didn't write the framework for this, but I'm not sure if it supports the windows version of the game, I'll look into it.

**Comment by cooldude315** [comment]
ok thanks

**Comment by AlexBurneikis** [comment]
You should try running The framework this is forked from and seeing if it renders ur ship

**Comment by cooldude315** [comment]
i had made a remake of it and had ships and it worked but idk why yours is not working for me, is it because im on win11, because when i tried the ships one i have it works but this isnt

**Comment by AlexBurneikis** [comment]
What resolution do you play at?

**Comment by cooldude315** [comment]
1920x1080

**Comment by AlexBurneikis** [comment]
Is the crosshair centred on your screen?

**Comment by cooldude315** [comment]
yeah

> Is the crosshair centred on your screen?

**Comment by AlexBurneikis** [comment]
Perhaps add a print statement within the sot_hack file during read actors and see if you reach it, and then within the ship part as well.

**Comment by cooldude315** [comment]
im not good with python can you show me where exactly?

**Comment by AlexBurneikis** [comment]
Try adding 
print("test") 
at line 183

**Comment by cooldude315** [comment]
it just does the same thing

**Comment by cooldude315** [comment]
but it fixes the problem where it doesnt want to load

**Comment by AlexBurneikis** [comment]
But does "test" show up in ur console

**Comment by cooldude315** [comment]
> But does "test" show up in ur console

yes

**Comment by AlexBurneikis** [comment]
Alrighty, remove that test line and add a new one here ![image](https://user-images.githubusercontent.com/87457198/200429251-27a98291-94ce-4c01-b4a8-273e5d88c2da.jpeg)

(Between 187-188)

**Comment by cooldude315** [comment]
yeah it shows 2 test every 5 seconds

**Comment by AlexBurneikis** [comment]
Maybe try line 73 in ship.py now, that way we can see if it is initialising the ship's class

**Comment by cooldude315** [comment]
ok

**Comment by cooldude315** [comment]
yes it does and it generates 7 every 3 sec 
![image](https://user-images.githubusercontent.com/117041200/200434605-e325d515-df0a-4672-b4fc-1b39ac826c28.png)

**Comment by AlexBurneikis** [comment]
Well, we know it's able to find the ship, there might be some sort of problem with the world to screen method

**Comment by jsim94** [comment]
You cut out the actual error in your screenshot, however if it's 
`pyglet.gl.lib.GLException: (0x502)` 
then go here > https://github.com/DougTheDruid/SoT-ESP-Framework/issues/16

**Comment by cooldude315** [comment]
> You cut out the actual error in your screenshot, however if it's `pyglet.gl.lib.GLException: (0x502)` then go here > [DougTheDruid#16](https://github.com/DougTheDruid/SoT-ESP-Framework/issues/16)

yep this fixed my issue, thank you both for the help 😃

**Comment by cooldude315** [comment]
oh and another question, does the canon aimbot is it using invalid angles (bannable angles) or valid angles?

**Comment by AlexBurneikis** [comment]
The cannon aimbot is only enabled at <=250m so it doesn't look Keep in mind that it currently only works if you are not moving, I will implement a moving method, but I'm on holiday rn and will do it when I get back.

**Comment by cooldude315** [comment]
but in the same time there are invalid angles so if you were aiming at one point and you snap onto it it could trigger something in the game to flag you

**Comment by cooldude315** [comment]
ive talk to people that have made some cheats and they say that using invalid angles in this game could get you banned

**Comment by AlexBurneikis** [comment]
My cheat, doesn't snap on to any angles, it just shows a circle to aim at.

**Comment by cooldude315** [comment]
oh cool my bad for not understand sorry

**Comment by AlexBurneikis** [comment]
No worries

**Comment by AlexBurneikis** [comment]
The ship prediction only works within 250m however the ship locations are visible as far as possible ~3000m

**Comment by cooldude315** [comment]
are you still working on the ship prediction?

**Comment by AlexBurneikis** [comment]
Yes. It currently only works while you are not moving, I will make it work while you are moving when I am home. I will be working on it again on Tuesday the 15th

**Comment by cooldude315** [comment]
btw can i give you some ideas that you might want to implement into your cheat?

**Comment by AlexBurneikis** [comment]
Sure thing

**Comment by cooldude315** [comment]
dig spots like X marks esp
crates esp
pointer to a location so if you put a pointer in the table map it will show where that is

**Comment by AlexBurneikis** [comment]
Unfortunately this cheat is only built for pvp, however I may make another cheat which I will include x marks and map pins. 
The ship speed is something that I will add when I'm back.

**Comment by cooldude315** [comment]
1 last idea and i think this one is good for pvp, a canon ball loader, auto puts the canon for you

**Comment by AlexBurneikis** [comment]
That is a good idea

**Comment by cooldude315** [comment]
for the aim predictions, it doesnt really calculate it correctly

**Comment by AlexBurneikis** [comment]
I know, as I have said, The cannon aim is incomplete and I will continue to work on it when I get back from my holiday in ~4 days

**Comment by cooldude315** [comment]
oh sorry, enjoy your holiday btw :D

**Comment by AlexBurneikis** [comment]
No worries, I will.
